include/picongpu/particles/Synchrotron/AlgorithmSynchrotron.hpp

```diff
@@ -0,0 +1,428 @@
+                    float_X const mass = attribute::getMass(parentElectron[weighting_], parentElectron);
+                    float3_X const vel = Velocity()(parentElectron[momentum_], mass);
+
+                    float_X const Vmag = pmacc::math::l2norm(vel);
+                    float3_X const crossVB = pmacc::math::cross(vel, bField);
+                    float3_X const Vnorm = vel / Vmag; // Velocity normalized -> not L2Norm
+                    float_X const dotVnormE = pmacc::math::dot(Vnorm, eField);
+                    float3_X const eFieldPlusCrossVB = eField + crossVB;
+
+                    float_X HeffValue = pmacc::math::dot(eFieldPlusCrossVB, eFieldPlusCrossVB) - dotVnormE * dotVnormE;
+                    if(HeffValue <= 0)
+                    {
+                        return 0;
+                    }
+                    HeffValue = math::sqrt(HeffValue);
+
+
+                    // Calculate chi
+                    float_X const gamma = Gamma()(parentElectron[momentum_], mass);
+                    float_X const chi = -ELECTRON_CHARGE * HBAR * HeffValue * gamma
+                        / (ELECTRON_MASS * ELECTRON_MASS * SPEED_OF_LIGHT * SPEED_OF_LIGHT * SPEED_OF_LIGHT);
+
+                    // zq
+                    float_64 oneMinusDeltaOverDelta = (1 - delta) / delta;
+                    float_X zq = 2. / (3 * chi) / oneMinusDeltaOverDelta;
+
+                    // zq convert to index and F1 and F2
+                    const float_X zqExponent = math::log2(zq);
+                    const int16_t index
+                        = static_cast<int16_t>((zqExponent - minZqExponent) / stepWidthLogatihmicScale);
+
+                    float_X F1 = 0;
+                    float_X F2 = 0;
+                    float_X Ftemp = 0;
+                    if(index >= 0 && index < interpolationPoints - 2)
+                    {
+                        float_X zq1 = math::pow(2, minZqExponent + stepWidthLogatihmicScale * (index));
+                        float_X zq2 = math::pow(2, minZqExponent + stepWidthLogatihmicScale * (index + 1));
+                        float_X f = (zq - zq1) / (zq2 - zq1);
+
+                        /// @todo Test Logarithmic interpolation
+                        F1 = F1F2(index, 0);
+                        Ftemp = F1F2(index + 1, 0);
+                        F1 = math::pow(F1, 1 - f) * math::pow(Ftemp, f); // F1 = F1**((1-f)) * Ftemp**f
+
+                        F2 = F1F2(index, 1);
+                        Ftemp = F1F2(index + 1, 1);
+                        F2 = math::pow(F2, 1 - f) * math::pow(Ftemp, f); // F2 = F2**((1-f)) * Ftemp**f
+                    }
+
+                    // Calculating the numeric factor: dt * (e**2 * m_e * c /( hbar**2 * eps0 * 4 * np.pi))
+                    float_X numericFactor = DELTA_T
+                        * (ELECTRON_CHARGE * ELECTRON_CHARGE * ELECTRON_MASS * SPEED_OF_LIGHT
+                           / (HBAR * HBAR * EPS0 * 4 * PI));
```

Comment:
here, you could reuse the previously defined `E_Schwinger`

Reply:
Never mind, keep it as in the paper and not use `E_Schwinger`

Reply:
ok